### PR TITLE
doc: zigbee: add error handler library page

### PR DESCRIPTION
**Pull request by greg-fer**
Added documentation page for Zigbee error handler lib.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: include/zigbee/zigbee_error_handler.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+You can configure the Zigbee error handler library using the :option:`CONFIG_ZBOSS_ERROR_PRINT_TO_LOG` Kconfig option.
+This option is enabled by default when you enable :ref:`zigbee_ug_logging` using the :option:`CONFIG_LOG`.
+The :option:`CONFIG_ZBOSS_ERROR_PRINT_TO_LOG` option automatically enables the :option:`CONFIG_ZIGBEE_ERROR_TO_STRING_ENABLED` Kconfig option, which obtains the ZBOSS error code that is supplied and printed to the log.
+
+API documentation
+*****************
+
+| Header file: :file:`include/zigbee/zigbee_error_handler.h`
+| Source file: n/a
```
Comment (sebastiandraus) [inline include/zigbee/zigbee_error_handler.rst:27]: Is this needed when empty?
Also, the `zb_error_to_string.c` in nrfxlib is a source file used by this API.

**Review by sebastiandraus — APPROVED** [review]
Just one question, otherwise looks good.

File: include/zigbee/zigbee_error_handler.rst (hunk 1 above)
Comment (greg-fer) [reply include/zigbee/zigbee_error_handler.rst:27]: OK, so I guess we need to describe in the overview that this library has the ZBOSS source file in nrfxlib, no? Not sure how many other libs have sources in a different submodule...

File: include/zigbee/zigbee_error_handler.h [hunk 2]
```diff
@@ -22,15 +19,18 @@
 #include <zboss_api.h>
 #include <zb_errors.h>
 
-#define ZB_ERROR_BASE_NUM 20000 /*< ZBOSS stack error codes base. */
+/**@brief ZBOSS stack error code base.
+ *
+ */
+#define ZB_ERROR_BASE_NUM 20000
```
Comment (tomchy) [inline include/zigbee/zigbee_error_handler.h:25]: Is this symbol used somewhere?

File: include/zigbee/zigbee_error_handler.h [hunk 3]
```diff
@@ -22,15 +19,18 @@
...
 
 #ifdef CONFIG_ZBOSS_ERROR_PRINT_TO_LOG
 #include "zb_error_to_string.h"
 #include <logging/log.h>
 
 
-/**@brief Macro for calling error handler function if supplied
- *        error code any other than RET_OK.
+/**@brief Macro for calling the error handler function if the supplied
+ *        error code is different than RET_OK.
```
Comment (tomchy) [inline include/zigbee/zigbee_error_handler.h:33]: `error code` -> `ZBOSS return code`?

File: include/zigbee/zigbee_error_handler.rst [hunk 4]
```diff
@@ -0,0 +1,31 @@
+.. _lib_zigbee_error_handler:
+
+Zigbee error handler
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Zigbee error handler library provides error codes for ZBOSS functions that return codes other than ``RET_OK``.
```
Comment (tomchy) [inline include/zigbee/zigbee_error_handler.rst:10]: No, it does not. The error codes are provided as a part of ZBOSS API.
This section has to be refactored.

File: include/zigbee/zigbee_error_handler.rst [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+It is used by default in the :ref:`zigbee_ug_logging_stack_logs` when you enable :ref:`zigbee_ug_logging`.
+Enabling this library allows you to use the error handling utilities for the :ref:`nrfxlib:zboss` stack when working with the :ref:`ug_zigbee` protocol in the |NCS|.
+
+.. _lib_zigbee_error_handler_options:
+
+Configuration
+*************
+
+You can configure the Zigbee error handler library using the :option:`CONFIG_ZBOSS_ERROR_PRINT_TO_LOG` Kconfig option.
```
Comment (tomchy) [inline include/zigbee/zigbee_error_handler.rst:19]: Configured or enabled?

**Review by tomchy — CHANGES_REQUESTED** [review]

File: include/zigbee/zigbee_error_handler.rst (hunk 1 above)
Comment (sebastiandraus) [reply include/zigbee/zigbee_error_handler.rst:27]: AFAIK the `zb_error_to_string.c` i not part of the ZBOSS stack.
@tomchy Do you remember if the `zb_error_to_string.c` needs to be in the `nrxlib`? Maybe we can move it to `nrf`.

File: include/zigbee/zigbee_error_handler.h (hunk 2 above)
Comment (greg-fer) [reply include/zigbee/zigbee_error_handler.h:25]: Dunno. Do we want to keep it?

Comment (tomchy) [reply include/zigbee/zigbee_error_handler.h:25]: No, we want to remove it.

**Review by tomchy — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**